src/lcgs/ir/symbol_checker.rs

```diff
@@ -186,4 +191,82 @@ impl<'a> SymbolChecker<'a> {
             ),
         })
     }
+    /// First combines all numbers, as we already know the min of that
+    /// Then returns a new checked Vec of Expr to find Min of.
+    fn check_min(&self, ls: &Vec<Expr>) -> Result<Expr, ()> {
+        let checked_list: Vec<Expr> = ls.iter().map(|p| self.check(p).unwrap()).collect();
+        let number: Option<i32> = checked_list
+            .iter()
+            .filter_map(|p| match p.kind {
+                ExprKind::Number(v) => Some(v),
+                _ => None,
+            })
+            .min();
+        return if let Some(x) = number {
+            let mut res: Vec<Expr> = checked_list
+                .into_iter()
+                .filter(|p| !matches!(p.kind, ExprKind::Number(_)))
+                .collect();
+            if res.is_empty() {
+                Ok(Expr {
+                    kind: ExprKind::Number(x),
+                })
+            } else {
+                res.push(Expr {
+                    kind: ExprKind::Number(x),
+                });
+                Ok(Expr {
+                    kind: ExprKind::Min(res),
+                })
+            }
+        } else {
+            Ok(Expr {
+                kind: ExprKind::Min(
+                    checked_list
+                        .iter()
+                        .map(|p| self.check(p).unwrap())
+                        .collect(),
```

Comment:
The expressions in `checked_list` are already checked. Same problem in `check_max`.

Reply:
5f29b4d